lib/std/fs.zig

```diff
@@ -2170,6 +2170,16 @@ pub const Dir = struct {
         return file.stat();
     }
 
+    pub const StatFileError = File.OpenError || StatError;
+
+    // TODO: improve this to use the fstatat syscall instead of making 2 syscalls here.
```

Comment:
Whenever `stat` involved, a reason about [TOCTOU](https://insanecoding.blogspot.com/2007/03/file-descriptors-and-why-we-cant-use.html) should be added.
A detailed reasoning of what weird things can happen, if one only uses `stat` before `open`: https://unparalleled.eu/publications/2021/advisory-unpar-2021-0.txt
So this should at least argue, why it is fine to use `stat` here.